### PR TITLE
C4F3 - Feature, record suggested synonym for a given word

### PR DESCRIPTION
**Pull request by omar-mohyeldin**
issue #83 
@AmrARaouf @karimngr @Mirna-Yacout @nohahesham Please review

### CODE REVIEW

File: arability/app/models/synonym.rb [hunk 1]
```diff
@@ -14,8 +13,20 @@ def self.find_loacle
     end
   end
 
-  validates_format_of :name, :with => /^([\u0621-\u0652 ])+$/,
-    :message => Synonym.find_loacle 
+  def existing?
```
Comment (AmrARaouf) [inline arability/app/models/synonym.rb:16]: where's the documentation for this function?

File: arability/app/models/synonym.rb [hunk 2]
```diff
@@ -131,20 +142,41 @@ def retrieve_synonyms(keyword)
```
Comment (AmrARaouf) [inline arability/app/models/synonym.rb:141]: where's the rest of the documentation?

File: arability/app/models/synonym.rb (hunk 1 above)
Comment (omar-mohyeldin) [inline arability/app/models/synonym.rb:16]: This is a helper function for the validation, just like the one above it for the locale, but has nothing to do with my stories

File: arability/app/models/synonym.rb (hunk 2 above)
Comment (omar-mohyeldin) [inline arability/app/models/synonym.rb:141]: I haven't changed anything in it, so github skipped it

**Comment by AmrARaouf** [comment]
doc and testing reviewed

File: arability/app/models/synonym.rb (hunk 2 above)
Comment (nohahesham) [inline arability/app/models/synonym.rb:141]: why there i a failure n then the rest of the code? where is the rest?

Comment (omar-mohyeldin) [inline arability/app/models/synonym.rb:141]: the rest of the code I haven't changed in it anything, so github automatically generates such statement to indicate that it had skipped the unchanged codes and jumped directly to codes modified by me

**Comment by nohahesham** [comment]
code reviewed:D

**Comment by Mirna-Yacout** [comment]
@Omar-Hossam RO1 reviewed
in RO2 check your return statement it should not be new_synonym model

**Comment by omar-mohyeldin** [comment]
@Mirna-Yacout UMLs modified, please review again

**Comment by Mirna-Yacout** [comment]
UML reviewed

**Comment by ghost** [comment]
good to go :+1:

**Pull request merged**